### DISCUSSION TITLE
Questions about thermal hydraulics module

### DISCUSSION THREAD

**Discussion by Raven-pro** [idaholab/moose#22763] [Q&A Modules: General] [answered]
Hello developers!

I am using the TH module to simulate the conjugate heat transfer process of the 1-D single-phase flow in vertical pipes.

As in tutorial 02, when implementing the conjugate heat transfer, the power is set to be constant everywhere (I also want to ask about what tot_power means in the input file by the way, does that mean power of the whole rod or just of that single element of the rod)

But I want to make the power distribution different, like individually setting the power of each node, Is there any way to achieve that?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You may specify the "power_shape_function" parameter in the HeatSourceFromTotalPower to have a shape.
This should be the name of a Function, which may have any spatial and time dependence. 

The total power is specified in a TotalPower component
the heat source component uses this TotalPower component to know how much power to generate

Guillaume
